src/lib/state-operators/removal.ts

```diff
@@ -0,0 +1,91 @@
+import { StateOperator } from '@ngxs/store';
+import { compose, patch } from '@ngxs/store/operators';
+import { Predicate } from '@ngxs/store/operators/internals';
+import { Dictionary } from '../internal';
+import { EntityStateModel } from '../models';
+import { updateTimestamp } from './timestamp';
+
+/**
+ * Removes all entities, clears the active entity and updates the `lastUpdated` timestamp.
+ */
+export function removeAllEntities<T>(): StateOperator<EntityStateModel<T>> {
+  return (state: EntityStateModel<T>) => {
+    return {
+      ...state,
+      entities: {},
+      ids: [],
+      active: undefined,
+      lastUpdated: Date.now()
+    };
+  };
+}
+
+/**
+ * Removes the entities specified by the given IDs.
+ * The active entity will be cleared if included in the given IDs.
+ * Updates the `lastUpdated` timestamp.
+ * @param ids IDs to remove
+ */
+export function removeEntities<T>(ids: string[]): StateOperator<EntityStateModel<T>> {
+  const entityRemoval = patch<EntityStateModel<any>>({
+    entities: removeEntitiesFromDictionary(ids),
+    ids: removeEntitiesFromArray(ids)
+  });
+  return compose(
+    entityRemoval,
+    clearActiveOnRemoval(ids),
+    updateTimestamp()
+  );
+}
+
+/**
+ * Only clears the `active` entity, if it's included in the given array.
+ * All other fields will remain untouched in any case.
+ * @param idsForRemoval the IDs to be removed
+ */
+export function clearActiveOnRemoval<T>(
+  idsForRemoval: string[]
+): StateOperator<EntityStateModel<T>> {
+  return (state: EntityStateModel<any>) => {
+    return {
+      ...state,
+      active: idsForRemoval.includes(state.active) ? undefined : state.active
+    };
+  };
+}
+
+/**
+ * Removes the given items from the existing items, based on equality.
+ * @param forRemoval items to remove
+ */
+export function removeEntitiesFromArray<T>(forRemoval: T[]): StateOperator<Array<T>> {
+  return (existing: ReadonlyArray<T>) => {
+    return existing.filter(v => !forRemoval.includes(v));
+  };
+}
+
+/**
+ * Removes all items from the existing items that match the given predicate.
+ * @param predicate predicate to determine if item should be removed
+ */
+export function removeEntitiesByPredicate<T>(
```

Comment:
You could use the `removeItem` operator that is built into ngxs instead of this.
See: https://github.com/ngxs/store/blob/71886fbcb4b032200093d32646d63526151535f4/packages/store/operators/src/remove-item.ts